diff --git a/deps/uv/uv.gyp b/deps/uv/uv.gyp
index 540445f1f32..043d8ca71c2 100644
--- a/deps/uv/uv.gyp
+++ b/deps/uv/uv.gyp
@@ -156,7 +156,7 @@
       'target_name': 'libuv',
       'toolsets': ['host', 'target'],
       'type': '<(uv_library)',
-      'include_dirs': [
+      'include_dirs+': [
         'include',
         'src/',
       ],
@@ -171,7 +171,7 @@
           '<@(shared_unix_defines)',
           '<@(shared_zos_defines)',
         ],
-        'include_dirs': [ 'include' ],
+        'include_dirs+': [ 'include' ],
         'conditions': [
           ['OS == "linux" or OS=="openharmony"', {
             'defines': [ '_POSIX_C_SOURCE=200112' ],
@@ -308,6 +308,7 @@
             '_XOPEN_SOURCE=500',
             '_REENTRANT',
           ],
+          'defines': [ '__USE_GNU' ],
           'link_settings': {
             'libraries': [
               '-lkstat',
